hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveClientPool.java

```diff
@@ -116,6 +119,52 @@ public void testGetTablesFailsForNonReconnectableException() throws Exception {
         .hasMessage("Another meta exception");
   }
 
+  @Test
+  public void testExceptionMessages() {
+    // Test Wrapped MetaException with a message
+    try (MockedStatic<MetaStoreUtils> mockedStatic = Mockito.mockStatic(MetaStoreUtils.class)) {
```

Comment:
Couldn't find a way to mock calling method 
```
return GET_CLIENT.invoke(
            hiveConf, (HiveMetaHookLoader) tbl -> null, HiveMetaStoreClient.class.getName());
```
and throw Exception. Had to trace and mock the class which is throwing error.